frog/imports/ui/App.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from 'react';
+
+import Body from './Body.jsx';
+import HeaderButton from './HeaderButton.jsx';
+
+// App component - represents the whole app
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    // The state of the App component indicates what the app is showing
+    // The display automatically updates itself on a call of setState()
+    // The Body component has app={this.state.app} as a property
+    this.state = {
+      app: "home",
+    };
+  }
+
+  renderButtons() {
```

Comment:
Here's an attempt at a rewrite using more pure components, and splitting out some of the logic:

```js
import React, { Component } from 'react';

import Body from './Body.jsx';
import HeaderButton from './HeaderButton.jsx';

const apps = ["home","repository","editor","engine","analytics"];

const AppButton = ({app, changeFn}) => { 
  const appobj = {
    id: app,
    onClick: () => changeFn(app)
  }
  return(
    <HeaderButton data={appobj}  />
  )
}

const Buttons = ({apps, changeFn}) => { 
  return(
    <div>
      {apps.map(app => <AppButton app={app} changeFn={changeFn} key={app}/>)}
    </div>
  )
}


// App component - represents the whole app
export default class App extends Component {
  constructor(props) {
    super(props);
    // The state of the App component indicates what the app is showing
    // The display automatically updates itself on a call of setState()
    // The Body component has app={this.state.app} as a property
    this.state = {
      app: "home",
    };
  }

  render() {
    console.log(this.state)
    return (
      <div>
        <div id="header">
          <h1>FROG - FABRICATING AND RUNNING ORCHESTRATION GRAPHS</h1>
          <Buttons apps={apps} changeFn={(app) => this.setState({app: app})}/>
        </div>
        <div id="body">
          <Body app={this.state.app} />
        </div>
      </div>
    );
  }
}
```